pandas/io/json/_normalize.py

```diff
@@ -230,15 +230,18 @@ def _json_normalize(
     Returns normalized data with columns prefixed with the given string.
     """
 
-    def _pull_field(js: Dict[str, Any], spec: Union[List, str]) -> Iterable:
+    def _pull_field(
+        js: Dict[str, Any], spec: Union[List, str], is_meta: bool = True
+    ) -> Any:
         result = js  # type: ignore
         if isinstance(spec, list):
             for field in spec:
                 result = result[field]
         else:
             result = result[spec]
 
-        if not isinstance(result, Iterable):
+        # GH 31507 iterable limit should only be used on record, not meta
```

Comment:
this is a code smell to add this. so it seems that result can be null, iterabe, or a scalar? if its a scalar what is the return value here?

Reply:
you mean `[99]`? then will still return 99

i think this `Iterable` should only restrict if specifying `record_path`, but not for `meta`? Am I right about the behaviour here @WillAyd ?

Reply:
I do agree that this is getting a little strange, especially since there is inspection of `meta` on line 273 of the same module and we are essentially repeating that here with a boolean indicator being manually supplied

@charlesdong1991 do you see a way to more logically order this function so we don't have to use this bool indicator?

Reply:
thanks for your reply! @WillAyd 

I will think about it a bit, i feel the patch for that `Iterable` in this `_pull_field` has code smell a bit because it is used for two cases which have different requirements

Reply:
> especially since there is inspection of meta on line 273 of the same module and we are essentially repeating that here with a boolean indicator being manually supplied

i just take a look again at the current codebase, seems that line 273 and onwards you referred is to validate/transform the key which is assigned to `meta` not the value the key associated with, in this case, `key` is `id` (and we could either specify it as `[id]` or `id` or more complex cases with nested list, and this part of code will deal with it), however, the code in `_pull_field` is to pull the value out, and in this case, the value is `99`, and for `record_path` the value should be an `Iterable`, while for `meta` it does not necessarily be the case, and therefore I think the patch added to check the type should only work for values of which the element of `record_path` point to, not for `meta`? @WillAyd

Reply:
@charlesdong1991 does splitting this into separate functions for record_path vs meta and just using this as a base for those functions make things cleaner?